code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -160,22 +160,27 @@ nobiliumsuppression = INFINITY
 	cached_results["fire"] = 0
 	var/turf/open/location = isturf(holder) ? holder : null
 	var/burned_fuel = 0
+
 	if(cached_gases[/datum/gas/oxygen][MOLES] < cached_gases[/datum/gas/tritium][MOLES] || MINIMUM_TRIT_OXYBURN_ENERGY > air.thermal_energy())
 		burned_fuel = cached_gases[/datum/gas/oxygen][MOLES] / TRITIUM_BURN_OXY_FACTOR
 		cached_gases[/datum/gas/tritium][MOLES] -= burned_fuel
+
+		ASSERT_GAS(/datum/gas/water_vapor, air) //oxygen+more-or-less hydrogen=H2O
+		cached_gases[/datum/gas/water_vapor][MOLES] += burned_fuel / TRITIUM_BURN_OXY_FACTOR
```

Comment:
Why move this up?

Reply:
Since burned_fuel is multiplied by 10 for saturated fires, it produces more water per reaction tick for saturated fires than unsaturated fires. This accounts for that

Reply:
but burned_fuel is only modified once, cause of the else. Isn't keeping it in the `if(burned_fuel)` block the same behavior? or am I missing something.

Reply:
the old if(burned_fuel) part divides burned_fuel by 100 and saturated burns multiplies it by 10. Since i removed the *10 for burned_fuel, keeping it there would make saturated fires produce 10 times less water vapor. By moving it into the if-else part, it keeps the old behavior of producing 10 times more water vapour for saturated fires per reaction tick.

Reply:
Ah I see